src/appSelector.ts

```diff
@@ -153,6 +237,9 @@ export function findAppInFolder(folderUri: vscode.Uri): App | undefined {
         found = false;
         break;
     }
+    if (appLanguage === "C") {
```

Comment:
How about moving this after the `if (found)` line so we are sure only call it if an app is really found ?